python/ray/data/tests/test_dataset_iceberg.py

```diff
@@ -0,0 +1,29 @@
+import ray
+import pytest
+import getpass as gt
+
+
+@pytest.mark.skipif(
+    gt.getuser() != "jonasjiang",
+    reason="A place holder for testing ray.data.read_iceberg api",
```

Comment:
This PR is currently lack of unit tests since `pyiceberg` does not support write/insert data to any iceberg table. So it would be a liitle hard to construct a samle iceberg table and test the api on that.

Recently, there are several ways I can come up with to test the API:
1. In pyiceberg, we includes a docker container in integration test pipeline to create some test tables in a local minio instance. https://github.com/apache/iceberg/blob/master/python/dev/docker-compose-integration.yml
Not sure if it is applicable or worthwhile to include the additional containers in ray's CI
2. Like: https://github.com/ray-project/ray/blob/76946cfad7d7e92a0efe7efb6ef5ba4d73945fea/python/ray/data/tests/test_dataset_formats.py#L216-L223
We can try to fetch an iceberg table from some public catalog services and perform the test. However, I do not know if there exist such public catalog or dataset in iceberg format.

Please feel free to share your ideas on how to test the `read_iceberg`. Thank you in advance for your contribution.